jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -817,6 +817,58 @@ ecma_builtin_helper_def_prop_by_index (ecma_object_t *obj_p, /**< object */
   return ret_value;
 } /* ecma_builtin_helper_def_prop_by_index */
 
+/**
+ * Helper function for at() functions.
+ *
+ * See also:
+ *          ECMA-262 Stage 3 Draft Relative Indexing Method proposal 3. 4. 5. 6.
+ *
+ * Used by:
+ *         - The Array.prototype.at routine.
+ *         - The String.prototype.at routine.
+ *         - The TypedArray.prototype.at routine.
+ *
+ * @return ECMA_VALUE_ERROR - on conversion error
+ *         ECMA_VALUE_UNDEFINED - if the requested index is not exist
+ *         ECMA_VALUE_EMPTY - otherwise
+ */
+ecma_value_t
+ecma_builtin_helper_calculate_index (ecma_value_t index, /**< relative index argument */
+                                     ecma_length_t length, /**< object's length */
+                                     ecma_length_t *out_index) /**< calculated index */
```

Comment:
We should add a an assert to make sure that the `out_index` is always a valid pointer (aka not null)